src/Microsoft.AspNetCore.Sockets.Http/Internal/Transports/LongPollingTransport.cs

```diff
@@ -30,10 +30,18 @@ public LongPollingTransport(CancellationToken timeoutToken, PipeReader applicati
         {
             try
             {
+                var headersFlushed = false;
+                if (context.Items.TryGetValue("FirstRequest", out var firstRequest))
```

Comment:
Why can't you just pass it into the constructor? The transport itself isn't long lived.

Reply:
Yeah, that works. We were dealing with the restriction of the `IHttpTransport` interface, but you're right, the ctor is invoked for each incoming poll.